File: go/problems/day16_test.go
package problems

import "testing"

var day16 Solvable[int] = Day16{}

func TestDay16_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day16_1")
	result := day16.Solve(input)

	var expected int = 46
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay16_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day16_2")
	result := day16.Solve(input)

	var expected int = 7067
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day16.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day16.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 16", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day16_1");
    const result = solve(input);
    assert.equal(result, 46);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day16_2");
    const result = solve(input);
    assert.equal(result, 7067);
  });
});
